### PR TITLE
fix: TOB-ZKTRIE-18 get_account can read past the buffer

### PR DESCRIPTION
**Pull request by vyzo**
Verify the slice length before blindnly casting and transmuting.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205044772140415

### CODE REVIEW

**Comment by vyzo** [comment]
fix is incorrect.

**Comment by vyzo** [comment]
Updated with the correct fix.

**Comment by vyzo** [comment]
rebased on main.

File: lib.go [hunk 1]
```diff
@@ -233,6 +233,29 @@ func TrieGet(p C.uintptr_t, key_c *C.uchar, key_sz C.int) unsafe.Pointer {
 	return C.CBytes(v)
 }
 
+// variant of TrieGet that specifies the expected value size for safety; if the expected value
```
Comment (Thegaram) [inline lib.go:236]:
```suggestion
// variant of TrieGet that specifies the expected value size for safety; if the actual value
```

**Review by Thegaram — APPROVED** [review]

**Review by noel2004 — APPROVED** [review]
it is ok

**Pull request merged**